tests/test_hrex_rbfe.py

```diff
@@ -129,15 +129,43 @@ def sample_and_record_rss(*args, **kwargs):
     # Initial permutation should be the identity
     np.testing.assert_array_equal(result.hrex_diagnostics.replica_idx_by_state_by_iter[0], np.arange(n_windows))
 
+    # Check that we can extract replica trajectories
+    n_atoms = result.final_result.initial_states[0].x0.shape[0]
+    rng = np.random.default_rng(2024)
+    n_atoms_subset = rng.choice(n_atoms)
+    atom_idxs = rng.choice(n_atoms, n_atoms_subset, replace=False)
+    trajs_by_replica = result.extract_trajectories_by_replica(atom_idxs)
+    assert trajs_by_replica.shape == (n_windows, md_params.n_frames, n_atoms_subset, 3)
+
+    # Check that the frame-to-frame rmsd is lower for replica trajectories versus state trajectories
+    def pairwise_rmsd(traj):
```

Comment:
[low-stakes naming adjustment] : `[rmsd(x[t], x[t+1]) for t in range(len(traj) - 1)]` seems more like `time_lagged_rmsd` or similar than `pairwise_rmsd`.

(low-stakes: in a test, and assertion intents are documented clearly in the next few lines)

(also, 👍 , this gives a clear and clever way to assert temporal coherence)